src/components/Layout/Header/NavBar/UserMenu.tsx

```diff
@@ -73,8 +90,12 @@ type WalletButtonProps = {
 } & Pick<InitialState, 'walletInfo'>
 
 const WalletButton = ({ isConnected, walletInfo, onConnect }: WalletButtonProps) => {
-  return isConnected ? (
-    <Button onClick={onConnect} leftIcon={<WalletImage walletInfo={walletInfo} />}>
+  return Boolean(walletInfo?.deviceId) ? (
+    <Button
+      onClick={onConnect}
+      leftIcon={<WalletImage walletInfo={walletInfo} />}
+      rightIcon={isConnected ? undefined : <WarningTwoIcon ml={2} w={3} h={3} color='yellow.500' />}
```

Comment:
Does this not work?
```
isConnected && <WariningTwoIcon />
```

Reply:
TypeScript says the value must be an Element or undefined. `false` is not valid.